python/weld/arrow_compat.pyx

```diff
@@ -0,0 +1,44 @@
+from pyarrow.lib cimport Array
+from libcpp.memory cimport shared_ptr, make_shared
+from libc.stdint cimport uint8_t, int64_t, uintptr_t
+from ctypes import cast, addressof
+
+from pyarrow.lib cimport pyarrow_wrap_array, DataType
+from pyarrow.includes.libarrow cimport *
```

Comment:
This is cool, I'm glad you got this working. We should think about some affordances in the public pyarrow Cython API to make this easier for you. I will have a closer look at the patch when I can